packages/client/hmi-client/src/components/workflow/tera-simulation-workflow.vue

```diff
@@ -111,8 +132,78 @@ function saveTransform(newTransform: { k: number; x: number; y: number }) {
 	canvasTransform = newTransform;
 }
 
+function createNewEdge(node: WorkflowNode, port: WorkflowPort) {
+	if (isCreatingNewEdge.value === false) {
+		newEdge.value = {
+			id: edges.value.length.toString(),
+			workflowId: '0',
+			points: [currentPortPosition, currentPortPosition],
+			source: node.id,
+			sourcePort: port,
+			target: undefined,
+			targetPort: undefined
+		};
+		isCreatingNewEdge.value = true;
+	} else if (newEdge.value) {
+		if (port.type === newEdge.value.sourcePort.type) {
+			newEdge.value.target = node.id;
+			newEdge.value.targetPort = port;
+			edges.value.push(newEdge.value);
+			cancelNewEdge();
+		} else {
+			cancelNewEdge();
+		}
+	}
+}
+
+function onCanvasClick() {
+	if (isCreatingNewEdge.value === true) {
+		cancelNewEdge();
+	}
+}
+
+function cancelNewEdge() {
+	isCreatingNewEdge.value = false;
+	newEdge.value = undefined;
+}
+
+function onPortMouseover(position: Position) {
+	currentPortPosition = position;
+}
+
+function mouseUpdate(event) {
+	if (newEdge.value && newEdge.value.points && newEdge.value.points.length === 2) {
+		mousePosition.value = {
+			x: (event.pageX - canvasTransform.x) / canvasTransform.k,
+			y: (event.pageY - 57 - canvasTransform.y) / canvasTransform.k
+		};
+		if (event.target.className === 'port') {
+			newEdge.value.points[1] = currentPortPosition;
+		} else {
+			newEdge.value.points[1] = mousePosition.value;
+		}
+	}
+}
+
+onMounted(() => window.addEventListener('mousemove', mouseUpdate));
+onUnmounted(() => window.removeEventListener('mousemove', mouseUpdate));
+
+function updateEdgePositions(node: WorkflowNode, { x, y }) {
+	edges.value.forEach((edge) => {
+		if (edge.source === node.id) {
+			edge.points[0].x += x / canvasTransform.k;
+			edge.points[0].y += y / canvasTransform.k;
+		}
+		if (edge.target === node.id) {
+			edge.points[1].x += x / canvasTransform.k;
+			edge.points[1].y += y / canvasTransform.k;
+		}
+	});
```

Comment:
Nitpick: You might want to change the first and last elements in `points`, just in case we have curved edges with more than 2-elements and someone doesn't get totally surprised.

Can probably move `x / canvasTransform.k;` and `y / canvasTransform.k` outside of the loop to be a tad more efficient.